File: Python/求n的阶乘.py
import sys
def factorial(x):
	if x<0:
		return x
	if x==0:
		return 1
	else :
		return x*factorial(x-1)
x=factorial(int(sys.argv[1]))
with open('result.txt','w') as fp:
	if x<0:
		fp.write('I')
	else :
		fp.write(str(x))

File: C/求n的阶乘.c
#include<stdio.h>
#include<stdlib.h>
int factorial(int x)
{
	if(x<0) return x;
	if(x==0) return 1;
	else return x*factorial(x-1);
}
int main(int argc,char** argv)
{
	FILE *fp;
	if((fp=fopen("result.txt","w"))==NULL)
		return -1;
	int ret;
	ret=factorial(atoi(argv[1]));
	if(ret<0)
		fprintf(fp,"I");
	else
		fprintf(fp,"%d",ret);
	fclose(fp);
	return 0;
}
